Update Process Workflow with Gherkin (#9)
sourcelevel/guidelines: resources/process_workflow.md

All SMART tasks should be linked to a User Story, which should be properly detailed with its goal (it can be written using the User Story pattern), Acceptance criteria, and BDD if necessary.

> We are following Gherkin BD workflow to describe our User Stories, so, we have a template in ClubHouse to be used:

Reviewer: **suggestion**: fix typo:

```suggestion
We are following Gherkin syntax to describe our User Stories:
```

There is no need to specify that its in `Clubhouse`

Author: Done in 7e85115

> *    Title: [Verb] [What] [Where]:

Reviewer: **suggestion:** add markdown in code format:

```
<!-- Template markdown here -->
```

Author: Done in b6c62c3

All SMART tasks should be linked to a User Story, which should be properly detailed with its goal (it can be written using the User Story pattern), Acceptance criteria, and BDD if necessary.

We are following Gherkin syntax to describe our User Stories:

### Title:

> ```

Reviewer: **suggestion:** you should add a linebreak after a Markdown title

Review other occurrences

Author: Done in Done in ba05e0e

[Verb] [What] [Where]

Examples:
- Fix typo in Dangerzone
- Investigate problems in GitLab Sync
- Add search field to repositories listing in Dangerzone
```

### Description:
```
- Scenario:
- Given
- And
- When
- Then

> 

Reviewer: **suggestion:** removing this additional empty line

Author: Done in ba05e0e
